Map Editor Support, Map Renderer Button and more!

CnCRAZER opened the pull request:
<img width="2559" height="1439" alt="image" src="https://github.com/user-attachments/assets/60e85fd8-3557-48a0-a7ac-43059e16b091" />

<img width="2559" height="1439" alt="image" src="https://github.com/user-attachments/assets/3158a764-14a5-4ee8-a753-ff8a817e8f4d" />


Introduces new map editor tools (WAE and FA2SP) and related resources, including executables, configuration, and help files.
WAE has been configured to read `cncnet.mix` before any `expandmdXX.mix` to ensure it reads our rules.

Relocates map renderer program within tools directory alongside WAE and FA2SP.
Adds `.bat` file to auto open custom maps folder when pressed.
Adds button to open up modenc with a browser.

Updates MainMenu.ini and ExtrasWindow.ini to add buttons for accessing these tools and extra resources.

CnCRAZER committed:
Add new map editor tools and update main menu

Introduces new map editor tools (WAE and FA2SP) and related resources, including executables, configuration, and help files.

Updates MainMenu.ini and ExtrasWindow.ini to add buttons for accessing these tools and extra resources. Also reorganizes the Map Renderer tools under a new directory and updates .gitignore accordingly.

CnCRAZER committed:
Update .gitignore and updateexec for FinalSun.ini handling

Removed FinalSun.ini from the ignored Map Editor directory and added it to the root .gitignore.

Also added FinalSun.ini to the updateexec file to ensure it is deleted

DeathFishAtEase commented:
> Mod Enc link button

What is the intention of adding this button? ModEnc is primarily about modification-related knowledge. Even some currently existing command-line content, we (ModEnc) are planning to migrate it. Therefore, for players, linking to ModEnc seems somewhat confusing. A better option is [C&C Wikia](https://cnc.fandom.com/), which is oriented towards all enthusiasts, and the game pages on ModEnc will also redirect to C&C Wikia in the future (in fact, as early as 2007, the two wikis could already interconnect through interwiki-linking, and many pages on ModEnc that mention vanilla game units directly add links targeting the corresponding pages on C&C Wikia like [here](https://modenc.renegadeprojects.com/NoSpawnAlt)).

If you have sufficient reasons to add a link to ModEnc's button, both Renegade and I agree that the silver pattern used in the bottom right corner of each page on ModEnc would be a good choice.

CnCRAZER commented:
> > Mod Enc link button
> 
> 
> 
> What is the intention of adding this button? ModEnc is primarily about modification-related knowledge. Even some currently existing command-line content, we (ModEnc) are planning to migrate it. Therefore, for players, linking to ModEnc seems somewhat confusing. A better option is [C&C Wikia](https://cnc.fandom.com/), which is oriented towards all enthusiasts, and the game pages on ModEnc will also redirect to C&C Wikia in the future (in fact, as early as 2007, the two wikis could already interconnect through interwiki-linking, and many pages on ModEnc that mention vanilla game units directly add links targeting the corresponding pages on C&C Wikia like [here](https://modenc.renegadeprojects.com/NoSpawnAlt)).
> 
> 
> 
> If you have sufficient reasons to add a link to ModEnc's button, both Renegade and I agree that the silver pattern used in the bottom right corner of each page on ModEnc would be a good choice.

I was just listing out ideas for conversational purposes. But thanks for that suggestion!

Metadorius commented:
> > Mod Enc link button
> 
> What is the intention of adding this button? 

Help for people who are looking into how to mod in custom maps.

CnCRAZER committed:
Merge branch 'Feature-Enhancements' of https://github.com/CnCNet/cncnet-yr-client-package into Feature-Enhancements

CnCRAZER committed:
Fix executable paths in Yuri Theme ExtrasWindow.ini

- Changed paths from forward slashes with 'Package/' prefix to backslash format
- WAE: Package/Tools/Map Editor/WAE/WorldAlteringEditor.exe → Tools\Map Editor\WAE\WorldAlteringEditor.exe
- FA2SP: Package/Tools/Map Editor/FA2SP/FA2SPLaunch.exe → Tools\Map Editor\FA2SP\FA2SPLaunch.exe
- Map Renderer: Package/Tools/Map Renderer/CNCMaps.Renderer.GUI.exe → Tools\Map Renderer\CNCMaps.Renderer.GUI.exe

These paths are now consistent with Soviet and Allied theme formats and properly relative to the package directory for correct resolution when launching executables.

CnCRAZER committed:
Add custom maps folder shortcut and update ExtrasWindow

Introduced OpenCustomMaps.bat to open the Custom Maps folder and updated ExtrasWindow.ini for all themes to add new buttons for accessing custom maps and the Mod Encyclopedia. Also updated .gitignore to exclude additional map editor log files.

CnCRAZER committed:
Update WAE to v1.7.7

https://github.com/CnCNet/WorldAlteringEditor/releases

CnCRAZER committed:
Support Ares/Phobos Include for WAE

CnCRAZER committed:
add cncnet.mix and stringtables to WAE

CnCRAZER committed:
Fix Encyclopedia spelling and fix soviet theme

CnCRAZER committed:
Add rules_cncnet.ini; update README & WAE

Introduce a new rules_cncnet.ini for community map-only rules and guidance (techlevel -1 and AIBuildThis=no) and add it to cncnet.pack and ra2mode.pack via [$Include]=rules_cncnet.ini. Update the cncnet community pack README to point to the game-assets path and instruct map authors to place custom rules in rules_cncnet.ini (WAE now supports reading it automatically). Increase WAE MaxWaypoint in Constants.ini to 2147483647 to support Phobos-scale waypoints.

CnCRAZER committed:
Add rules_cncnet.ini and new buildable bunker from McPwny

CnCRAZER commented:
<img width="562" height="347" alt="image" src="https://github.com/user-attachments/assets/1de65fde-84a1-45e7-9c98-7ede44fb897d" />

Added new `rules_cncnet.ini` to include from. All new additions will be unbuildable by players AND Ai.

WAE will show these new additions by default in the side bar from the screenshot above. 
Mappers now just need to put the section in the mapfile and change techlevel.

CnCRAZER committed:
Merge branch 'develop' into Feature-Enhancements
